src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -253,20 +253,33 @@ public async ValueTask<object> ReceiveOneAsync(TimeSpan? max = null, Cancellatio
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to receive message from TestActor queue. Will not wait.");
-                var didTake = _testState.Queue.TryTake(out var item, cancellationToken);
+                var didTake = _testState.Queue.Reader.TryRead(out var item);
                 take = (didTake, item);
             }
             else if (maxDuration.IsPositiveFinite())
             {
                 ConditionalLog(shouldLog, "Trying to receive message from TestActor queue within {0}", maxDuration);
-                take = await _testState.Queue.TryTakeAsync((int)maxDuration.TotalMilliseconds, cancellationToken)
-                    ;
+                var delayTask = Task.Delay(maxDuration, cancellationToken);
+                var readTask = _testState.Queue.Reader.WaitToReadAsync(cancellationToken).AsTask();
+                var completedTask = await Task.WhenAny(readTask, delayTask);
```

Comment:
Have to do this instead of playing around with `CancellationToken`s expiring, because otherwise those work via throwing exceptions. This allows us to gracefully determine if we've hit the next point because stuff was ready or because time expired.